### PR TITLE
Libc++ : add support for exceptions and rtti

### PR DESCRIPTION
**Pull request by manoharahk**
- Below patchset enables -fexpection & -frtti support in libc++
 - It links libsupc++ ABI to get run-time type information (RTTI) and exception handling functionality

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

File: build/configs/artik053/cxxtest/Make.defs [hunk 1]
```diff
@@ -56,6 +56,12 @@ include ${TOPDIR}/arch/arm/src/armv7-r/Toolchain.defs
 
 LDSCRIPT = flash.ld
 
+ifeq ($(CONFIG_LIBCXX_HAVE_LIBSUPCXX),y)
```
Comment (sunghan-chang) [inline build/configs/artik053/cxxtest/Make.defs]: need to check

File: build/configs/artik053/helloxx/Make.defs [hunk 2]
```diff
@@ -56,6 +56,12 @@ include ${TOPDIR}/arch/arm/src/armv7-r/Toolchain.defs
 
 LDSCRIPT = flash.ld
 
+ifeq ($(CONFIG_LIBCXX_HAVE_LIBSUPCXX),y)
```
Comment (sunghan-chang) [inline build/configs/artik053/helloxx/Make.defs]: same as previous

**Comment by sunghan-chang** [comment]
@manoharahk @pradeepns85 Is this same as #1245 ? Requested by JavisJang?

**Comment by manoharahk** [comment]
No. It is different.
#1245 is for missing API's.
This PR is for supporting exceptions & RTTI

**Comment by juitem** [comment]
@sunghan-chang, please review this commits

**Comment by jarvisjang** [comment]
I confirmed, this PR is okay.

**Comment by sunghan-chang** [comment]
@manoharahk Please resolve conflict.

**Comment by manoharahk** [comment]
Okay, I will rebase.

**Pull request merged**